packages/ingest/src/lock.ts

```diff
@@ -31,30 +43,54 @@ async function checkOldLocks(bucket, locks = []) {
 * @param {string} providerName - The provider name
 * @returns {integer} - Number of current locks in the bucket
 **/
-async function countLock(bucket, providerName) {
+export async function countLock(
+  bucket: string,
+  providerName: string
+): Promise<number> {
   const s3Objects = await listS3ObjectsV2({
     Bucket: bucket,
     Prefix: `${lockPrefix}/${providerName}`
   });
-  return checkOldLocks(bucket, s3Objects);
+
+  if (s3Objects === undefined) return 0;
+
+  const x = <Lock[]>s3Objects.filter(({ Key }) => Key !== undefined);
```

Comment:
The list operation returns objects where key is undefined?

Also, let's change `x` to `oldLocks`

Reply:
> The list operation returns objects where key is undefined?

Apparently? AWS created the TS definitions, and it says it can be undefined, so I guess so. I've never seen it in the wild, but I guess it could happen if they say so.